packages/teams-js/src/internal/communication.ts

```diff
@@ -320,6 +324,45 @@ export function sendMessageToParentAsync<T>(actionName: string, args: any[] | un
   });
 }
 
+/**
+ * @hidden
+ * Send a message to parent requesting a MessageChannel Port.
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function requestPortFromParentWithVersion(
+  apiVersionTag: string,
+  actionName: string,
+  args: any[] | undefined = undefined,
+): Promise<MessagePort> {
+  if (!isFollowingApiVersionTagFormat(apiVersionTag)) {
+    throw Error(
+      `apiVersionTag: ${apiVersionTag} passed in doesn't follow the pattern starting with 'v' followed by digits, then underscore with words, please check.`,
+    );
+  }
+  return new Promise((resolve) => {
+    const request = sendMessageToParentHelper(apiVersionTag, actionName, args);
+    resolve(waitForPort(request.id));
+  });
+}
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function waitForPort(requestId: number): Promise<MessagePort> {
+  return new Promise<MessagePort>((resolve, reject) => {
+    CommunicationPrivate.portCallbacks[requestId] = (port: MessagePort | undefined, args?: any[]) => {
+      if (port instanceof MessagePort) {
+        resolve(port);
+      } else {
+        // First arg is the error message, if present
+        reject(args && args.length > 0 ? args[0] : new Error('Host responded without port or error details.'));
```

Comment:
>reject(args && args.length > 0 ? args[0] : new Error('Host responded without port or error details.')); [](http://example.com/codeflow?start=8&length=103)

It seems... notable? that if someone wanted to handle the error from here, they'd have to be prepared to handle two different types of error patterns: either the promise rejection will contain a string or it will contain an `Error` object containing a string. Is this an important distinction for the caller to make (i.e., should they take different actions depending on which type of error pattern is being used)?

You are clearly handling this in your unit tests, so it's obviously possible to handle it (I also tried writing some [sample code](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAtgTwAoCcRwJYQKYwLwxjYDuMq6W2APGAK5wBG2KAfABRsrYQgA2AbtgA0MLgCtswKAEp8LGAG8AsACgYo7BKlsipAKIo0KNgCIAZiBAnp0gNyqAvndWqAhhARhgMM7S9QMcBgAExAAFQALDDAAczZpAC4yNEwcKn4QDGD5ZTUYKBQERVV1dVBIWC4IWl5YAldiVwxYRHJU7Hs8svAeXmwAOl4QOIADACVuGqgkgBIFKqmHEec8hxhgVyhgCJg2bCS-AGswEGIwWVzSmAwzXagEAAdsEDM92TwPmAByaBRomK+FxKV3WPT4AyGowAwq5aDEIrBmEZZgpsEsVlc1theDhrrc9tcKq4vM9bgYjECuqVyr0IcM2CMYXCETAkSAUCiBnBuBBXDE0ctOpjgQ5HC4VKFIv94rYgA) in the playground just to see how each case would be handled)

Anyway I just wondered whether this is a distinction without a difference and the "Host responded without port or error details." string should just be plainly in the rejection without being wrapped in an `Error` object, since in all these cases the caller is.... likely just going to print/log the message? Maybe you aren't really expecting the caller to do handle this rejection anyway?

Anyway I'm not sure I'm recommending anything change here but this did make me stop and think. Which I guess is a good thing to do on a Friday!

Reply:
I'm definitely not clear on the correct 'promise rejection' pattern for use in the SDK, or what callers should expect in terms of error handling. :)

Here we have an example I looked at for throwing/rejecting with the error payload from args:
<img width="1010" alt="image" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/56459261/c20b3fcc-fccb-4256-93ce-f2a26ecef6e0">

In other places in communication.ts, we are throwing 'normal' new Errors:
<img width="708" alt="image" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/56459261/3908de57-ae06-474d-9187-21d9eaf0c1e1">

I admit I was disappointed to find that the 'error from host' didn't fit the 'proper' error format and needed special handling, but it contains more useful information generally that the locally thrown ones.

Thoughts?

Reply:
I don't really feel strongly one way or another about this and I don't know what kind of precedent @TrevorJoelHarris wants to set going forward so I'll just sign-off from my perspective and defer to him if he wants to do any follow-up here.

Reply:
I definitely prefer option 1, but also accept that our codebase isn't currently very consistent about how errors are handled. I prefer wrapping the string in an error when possible rather than just rejecting with a plain string.

Long term, I'd like all of the errors to be combination code and message, so if throwing something like that here for the second case is possible that would be great. I won't hold up the PR for it though, given the sorry state we're in right now when it comes to error consistency. @kerrynf I will leave it up to you: if you don't mind throwing more consistent errors out of this function as part of this PR, that would be great! Every little bit helps.

Reply:
Pasting my message image here, since I crashed the page earlier and it didn't save... TLDR: I think this PR should complete as is, but we should tackle error handling ASAP. :)

<img width="568" alt="image" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/56459261/a4948e30-81ad-4ad7-8b94-1800fe4755e5">